pkg/identity/identity.go

```diff
@@ -106,20 +108,59 @@ type RoleCertificate struct {
 	X509Certificate string
 }
 
+type Token interface {
+	Domain() string
+	Role() string
+	Raw() string
+	Expiry() int64
+}
+
 // RoleToken stores role token
 type RoleToken struct {
-	Domain      string
-	Role        string
-	TokenString string
-	Expiry      int64
+	domain string
+	role   string
+	raw    string
```

Comment:
@WindzCUHK Seems like there is a better way than `raw` for the string token.
Previous `TokenString` sounds more natural to me.